.github/workflows/sync-gitster-git.yml

```diff
@@ -0,0 +1,120 @@
+name: sync-gitster-git-branches
+
+on:
+  schedule:
+    - cron: '17 6 * * *'
+  workflow_dispatch:
+
+env:
+  SOURCE_REPOSITORY: gitster/git
+  TARGET_REPOSITORY: gitgitgadget/git
+
+jobs:
+  sync-gitster-git-branches:
+    runs-on: ubuntu-latest
+    steps:
+      - name: check which refs need to be synchronized
+        uses: actions/github-script@v6
+        id: check
+        with:
+          script: |
+            const sleep = async (milliseconds) => {
+              return new Promise(resolve => setTimeout(resolve, milliseconds))
+            }
+
+            const getRefs = async (repository) => {
+              for (let i = 1; i <= 10; i++) {
+                try {
+                  const [owner, repo] = repository.split('/')
+                  const { data } = await github.rest.git.listMatchingRefs({
+                    owner,
+                    repo,
+                    ref: 'heads/'
+                  })
+                  return data
+                    .filter(e => e.ref.match(/^refs\/heads\/(maint-\d|[a-z][a-z]\/)/))
+                    .sort((a, b) => a.ref.localeCompare(b.ref))
+                } catch (e) {
+                  if (e?.status !== 502) throw e
+                }
+
+                const seconds = i * i + 15 * Math.random()
+                core.info(`Encountered a Server Error; retrying in ${seconds} seconds`)
+                await sleep(1000 * seconds)
+              }
```

Comment:
Should an error be thrown if the for loop completes?  ie, 10 failures occurred.

Reply:
Yes, of course! D'oh.